{
    "GSD": {
        "alias": "CVE-2023-28859",
        "id": "GSD-2023-28859"
    },
    "namespaces": {
        "cve.org": {
            "CVE_data_meta": {
                "ASSIGNER": "cve@mitre.org",
                "ID": "CVE-2023-28859",
                "STATE": "PUBLIC"
            },
            "affects": {
                "vendor": {
                    "vendor_data": [
                        {
                            "product": {
                                "product_data": [
                                    {
                                        "product_name": "n/a",
                                        "version": {
                                            "version_data": [
                                                {
                                                    "version_value": "not down converted",
                                                    "x_cve_json_5_version_data": {
                                                        "versions": [
                                                            {
                                                                "status": "unknown",
                                                                "version": "n/a"
                                                            }
                                                        ]
                                                    }
                                                }
                                            ]
                                        }
                                    }
                                ]
                            },
                            "vendor_name": "n/a"
                        }
                    ]
                }
            },
            "data_format": "MITRE",
            "data_type": "CVE",
            "data_version": "4.0",
            "description": {
                "description_data": [
                    {
                        "lang": "eng",
                        "value": "redis-py before 4.4.4 and 4.5.x before 4.5.4 leaves a connection open after canceling an async Redis command at an inopportune time, and can send response data to the client of an unrelated request. (This could, for example, happen for a non-pipeline operation.) NOTE: the solutions for CVE-2023-28859 address data leakage across AsyncIO connections in general."
                    }
                ]
            },
            "problemtype": {
                "problemtype_data": [
                    {
                        "description": [
                            {
                                "lang": "eng",
                                "value": "n/a"
                            }
                        ]
                    }
                ]
            },
            "references": {
                "reference_data": [
                    {
                        "name": "https://github.com/redis/redis-py/pull/2641",
                        "refsource": "MISC",
                        "url": "https://github.com/redis/redis-py/pull/2641"
                    },
                    {
                        "name": "https://github.com/redis/redis-py/issues/2665",
                        "refsource": "MISC",
                        "url": "https://github.com/redis/redis-py/issues/2665"
                    },
                    {
                        "name": "https://github.com/redis/redis-py/releases/tag/v4.4.4",
                        "refsource": "MISC",
                        "url": "https://github.com/redis/redis-py/releases/tag/v4.4.4"
                    },
                    {
                        "name": "https://github.com/redis/redis-py/releases/tag/v4.5.4",
                        "refsource": "MISC",
                        "url": "https://github.com/redis/redis-py/releases/tag/v4.5.4"
                    },
                    {
                        "name": "https://github.com/redis/redis-py/pull/2666",
                        "refsource": "MISC",
                        "url": "https://github.com/redis/redis-py/pull/2666"
                    }
                ]
            }
        },
        "nvd.nist.gov": {
            "configurations": {
                "CVE_data_version": "4.0",
                "nodes": []
            },
            "cve": {
                "CVE_data_meta": {
                    "ASSIGNER": "cve@mitre.org",
                    "ID": "CVE-2023-28859"
                },
                "data_format": "MITRE",
                "data_type": "CVE",
                "data_version": "4.0",
                "description": {
                    "description_data": [
                        {
                            "lang": "en",
                            "value": "redis-py before 4.4.4 and 4.5.x before 4.5.4 leaves a connection open after canceling an async Redis command at an inopportune time, and can send response data to the client of an unrelated request. (This could, for example, happen for a non-pipeline operation.) NOTE: the solutions for CVE-2023-28859 address data leakage across AsyncIO connections in general."
                        }
                    ]
                },
                "problemtype": {
                    "problemtype_data": [
                        {
                            "description": []
                        }
                    ]
                },
                "references": {
                    "reference_data": [
                        {
                            "name": "https://github.com/redis/redis-py/pull/2641",
                            "refsource": "MISC",
                            "tags": [],
                            "url": "https://github.com/redis/redis-py/pull/2641"
                        },
                        {
                            "name": "https://github.com/redis/redis-py/issues/2665",
                            "refsource": "MISC",
                            "tags": [],
                            "url": "https://github.com/redis/redis-py/issues/2665"
                        },
                        {
                            "name": "https://github.com/redis/redis-py/releases/tag/v4.4.4",
                            "refsource": "MISC",
                            "tags": [],
                            "url": "https://github.com/redis/redis-py/releases/tag/v4.4.4"
                        },
                        {
                            "name": "https://github.com/redis/redis-py/releases/tag/v4.5.4",
                            "refsource": "MISC",
                            "tags": [],
                            "url": "https://github.com/redis/redis-py/releases/tag/v4.5.4"
                        },
                        {
                            "name": "https://github.com/redis/redis-py/pull/2666",
                            "refsource": "MISC",
                            "tags": [],
                            "url": "https://github.com/redis/redis-py/pull/2666"
                        }
                    ]
                }
            },
            "impact": {},
            "lastModifiedDate": "2023-03-30T22:15Z",
            "publishedDate": "2023-03-26T19:15Z"
        }
    }
}